btsieve/tests/test_data/lib.sh

```diff
@@ -0,0 +1,58 @@
+container="" # Set by docker_run
+
+rm_file_if_exists() {
+    local file=$1
+
+    if [ -f $file ] ; then
+        rm $file
+    fi
+}
+
+docker_run() {
+    container=$(docker run -d -v /tmp/bitcoin:/root/.bitcoin coblox/bitcoin-core -regtest)
+    echo "Create container $container ... sleeping"
+    sleep 2
+}
+
+docker_stop() {
+    docker stop $container > /dev/null
+    echo "Stopped container $container"
+}
+
+docker_rm() {
+    docker rm $container > /dev/null
+    echo "Removed container $container"
+}
+
+docker_start() {
+    docker start $container > /dev/null
+    echo "Started container $container ... sleeping"
+    sleep 2
+}
+
+
+generate_101_blocks() {
+    docker exec $container bitcoin-cli -regtest generate 101 > /dev/null
+    echo "Generated 101 blocks"
+}
+
+generate_block() {
+    local path="$1"
+
+    docker exec $container bitcoin-cli -regtest generate 1 > /dev/null
+    blockhash=$(docker exec $container bitcoin-cli -regtest getbestblockhash)
```

Comment:
Save time by removing this line and instead use the result returned by the generate (it gives you the new block hash)

Reply:
It gives you an array containing one blockhash though. I was told to avoid dealing with arrays in bash...

Reply:
I've been trying to use the array returned instead and it is not working lol

Reply:
use jq :)

Reply:
Yeah could use jq. Or just leave it like that and have less dependencies :)

It is already working so ...